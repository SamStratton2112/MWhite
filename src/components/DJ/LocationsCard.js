import { Col, Container} from 'react-bootstrap';
import {useState} from 'react';
import List from '../List';

const LocationCard = ({location}) => {
    const [hidden, setHidden] = useState(true);

    let body1 = <img src={location.image}  style={{ width: '19rem', height: '19rem' }} alt={location.city} />;
    
    let list = Array.from(location.locations)
    let body2 = <List list={list} />
    return (
        <Col>
            <Container 
                style={{ width: '20rem', height: '20rem', border: 'solid' }} 
                className="venueCard" 
                onMouseEnter={() => setHidden(false)} 
                onMouseLeave={() => setHidden(true)}
            >
                <p>{location.city}</p>
                {hidden ? body1 : body2}
            </Container>
        </Col>
    );
};

export default LocationCard;