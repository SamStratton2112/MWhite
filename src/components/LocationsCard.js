import { Col, Container} from 'react-bootstrap';
import {useState} from 'react';
import List from './List';

const LocationCard = ({location}) => {
    const [hidden, setHidden] = useState(true);

    let body1 = <img src={location.image}  style={{ width: '19rem', height: '19rem' }} alt="coming soon" />;
    
    let list = Array.from(location.locations)
    let body2 = <List list={list} />
    return (
        <Col>
            <Container 
                style={{ width: '20rem', height: '20rem', border: 'solid' }} 
                className="venueCard" 
                onMouseEnter={() => setHidden(false)} 
                onMouseLeave={() => setHidden(true)}
            >
                {hidden ? body1 : body2}
            </Container>
        </Col>
    );
};

export default LocationCard;