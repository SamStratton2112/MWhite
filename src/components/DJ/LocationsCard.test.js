import LocationsCard from './LocationsCard';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

const testLocation = {
    image: "testImage",
    city: "testCity",
    locations: ["test item 1", "test item 2"]
}

describe(LocationsCard, () => {
    it("renders without crashing", () => {
        render(<MemoryRouter>
            <LocationsCard  location={testLocation}/>
        </MemoryRouter>);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<MemoryRouter>
            <LocationsCard location={testLocation} />
        </MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });
});