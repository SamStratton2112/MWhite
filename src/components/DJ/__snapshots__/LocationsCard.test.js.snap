// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    <p
      style="border-radius: .2em; margin-top: 1em; width: 12em;"
    >
      testCity
    </p>
    <div
      class="container"
      style="width: 20rem; height: 20rem;"
    >
      <img
        alt="testCity"
        src="testImage"
        style="width: 20rem; height: 20rem; border-radius: .5em;"
      />
    </div>
  </div>
</DocumentFragment>
`;
